dhall-lsp-server build failure with text 2.1.2 (using GHC 9.10.1)

When building `dhall-lsp-server` with GHC 9.10.1 and Cabal 3.14.1.1, it looks like `text 2.1.2` can get picked which causes error from ambiguous show (https://hackage.haskell.org/package/text-2.1.2/docs/Data-Text.html#g:27):
```
src/Dhall/LSP/Util.hs:16:16: error: [GHC-87543]
    Ambiguous occurrence ‘show’.
    It could refer to
       either ‘Prelude.show’,
              imported from ‘Prelude’ at src/Dhall/LSP/Util.hs:3:8-21
              (and originally defined in ‘ghc-internal-9.1001.0:GHC.Internal.Show’),
           or ‘Data.Text.show’,
              imported from ‘Data.Text’ at src/Dhall/LSP/Util.hs:11:1-16.
   |
16 | tshow = pack . show
   |                ^^^^
```

A workaround of `--constraint=text<2.1.2` allows build to finish.

I think fix would be fixing ambiguity with qualifying `show` (to `Prelude.show`?) or qualified import of `Data.Text` as recommended by documentation.

---

Will note that building with GHC 9.10.1 does require additional workarounds:
* in main branch, need to avoid resolving dhall-nix/dhall-nixpkgs if using cabal.project
* for Hackage release, need to work around lack of `dhall-json` release with https://github.com/dhall-lang/dhall-haskell/commit/28d346f00d12fa134b4c315974f76cc5557f1330 + https://github.com/dhall-lang/dhall-haskell/commit/277d8b1b3637ba2ce125783cc1936dc9591e67a7, and backporting https://github.com/dhall-lang/dhall-haskell/commit/a621e1438df5865d966597e2e1b0bb37e8311447

--- a/dhall-lsp-server/src/Dhall/LSP/Util.hs
+++ b/dhall-lsp-server/src/Dhall/LSP/Util.hs
@@ -7,26 +7,28 @@ module Dhall.LSP.Util (
   unlines'
 ) where
 
-import Data.List.NonEmpty
-import Data.Text
+import Data.List.NonEmpty (NonEmpty (..))
+import Data.Text (Text)
+
+import qualified Data.Text as Text
 
 -- | Shorthand for @pack . show@. Useful since we are mostly working with Text
 --   rather than String.
 tshow :: Show a => a -> Text
-tshow = pack . show
+tshow = Text.pack . show
 
 -- | A variant of @Data.Text.lines@ that does not swallow the last empty. Always
 --   returns at least the empty line!
 lines' :: Text -> NonEmpty Text
 lines' text =
-    case split (== '\n') text of
+    case Text.split (== '\n') text of
       []     -> "" :| []  -- this case never occurs!
       l : ls -> l  :| ls
 
 -- | A variant of @Data.Text.unlines@ that is the exact inverse to @lines'@ (and
 --   vice-versa).
 unlines' :: [Text] -> Text
-unlines' = intercalate "\n"
+unlines' = Text.intercalate "\n"
 
 rightToMaybe :: Either a b -> Maybe b
 rightToMaybe (Right b) = Just b

